test/shmemx/c11_shmemx_team_reduce.c

```diff
@@ -0,0 +1,260 @@
+          break;                                                               \
+      case xor:                                                                \
+          CHECK_DEST_BUFFER(OP, TYPE, (TYPE)(npes % 2 ? 1ULL : 0ULL));         \
+          break;                                                               \
+      case max:                                                                \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, 1ULL, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, 1ULL);                               \
+          break;                                                               \
+      case min:                                                                \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, 1ULL, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, 1ULL);                               \
+          break;                                                               \
+      case sum:                                                                \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, npes, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, npes);                               \
+          break;                                                               \
+      case prod:                                                               \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, 1ULL, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, 1ULL);                               \
+          break;                                                               \
+      default:                                                                 \
+          printf("Invalid operation (%d)\n", OP);                              \
+          shmem_global_exit(1);                                                \
+    }                                                                          \
+  } while (0)
+
+#else
+#define TEST_SHMEM_REDUCE(OP, TYPE)
+#endif
+
+
+int main(void) {
+
+    shmem_init();
+
+    int rc = EXIT_SUCCESS;
+
+    const int mype = shmem_my_pe();
+    const int npes = shmem_n_pes();
+
+    if (npes > MAX_NPES) {
+        if (mype == 0)
+            fprintf(stderr, "ERR - Requires less than %d PEs\n", MAX_NPES);
+        shmem_finalize();
```

Comment:
Should we use `shmem_global_exit` here? It seems odd to error'ed out without returning an error.

Reply:
We require unit tests to exit successfully when they don't support the given number of PEs.  Using `shmem_finalize` is a more reliable way of getting this behavior.